src/util/splitAtSelection.ts

```diff
@@ -0,0 +1,44 @@
+/**
+ * Split given root node into two different ranges at the given selection.
+ */
+function splitNode(selection: Selection, root: HTMLElement) {
+  const range = selection.getRangeAt(0)
+  const { firstChild, lastChild } = root
+
+  if (!firstChild || !lastChild) return
+
+  const previousRange = document.createRange()
+  previousRange.setStartBefore(firstChild)
+  previousRange.setEnd(range.startContainer, range.startOffset)
+
+  const nextRange = document.createRange()
+  nextRange.setStart(range.endContainer, range.endOffset)
+  nextRange.setEndAfter(lastChild)
+
+  return {
+    previous: previousRange,
+    next: nextRange,
+  }
```

Comment:
Let's use left and right instead of previous and next for consistency.

Reply:
I missed it here. I will change it.